src/rock/models/feeds/queries.js

```diff
@@ -4,7 +4,6 @@ export default [
 
   `userFeed(
     filters: [String]!
-    options: String = "{}"
```

Comment:
@Lepozepo I didn't catch this in the review, but we can't remove arguments like this from the GraphQL schema less we break all existing clients. So we need to add this back in.

See http://graphql.org/learn/best-practices/#versioning

Might be worth looking at something like https://github.com/entria/graphql-find-breaking-changes-cli in the future.

Reply:
@conrad-vanl it may be a bit more complex to maintain compatibility considering that not all of the previous clients’ data is coming from GraphQL. We could do it but I think we’ll need to either force update the old client to understand the new DB structure (easy to do because of cordova) or write something to keep the migrated collections in sync with the old structure.

Reply:
> We could do it but I think we’ll need to either force update the old client to understand the new DB structure (easy to do because of cordova) or write something to keep the migrated collections in sync with the old structure.

Expand on this some more. GraphQL is a living api schema, and so breaking changes _cannot_ be implemented. If we have other breaking changes in this branch right now, we need to roll them back.

Fields can be marked `deprecated` and they are removed from documentation, but we still need to support old fields. Query params like this - need to still be a part of the schema. Unfortunately, this attribute (`options`) being a JSON string was probably a not-too-great design decision for graphql, but we need to have some method of supporting it so that old clients aren't hit with errors (force updating old clients isn't a suitable option). Right now if you were to point Holtzman at alpha, it would crash.

Reply:
So, maybe a potential solution could be to keep options in the schema, but it is ignored...I don't like that idea as that is still altering the data existing clients receive, but it won't cause them to crash 🤔

Reply:
@conrad-vanl yes, we can add `options` and prevent the crash that way for sure. What the resolver does to resolve doesn't matter anyway so the extra info coming down the pipeline has no impact.

Reply:
My concern is that things like `likes` and `following` now exist in different mongo collections. These mongo collections don't have publishers in Holtzman so Holtzman will fail unless we sync with the legacy stuff or update Holtzman.

Reply:
But considering Holtzman is a cordova app, it doesn't matter. Old clients will automatically be upgraded to Apollos anyway and use all the new stuff.

Reply:
Also, how exactly did the DB structure change?

Reply:
@conrad-vanl following topics don't exist as an array under the users collection (they have their own collection). Likes didn't change much, the `userId` uses the `PrimaryAliasId` now instead of meteor's user collection _id.

Reply:
But both are breaking changes for Meteor pub/sub (as all user ID's have effectively changed to `PrimaryAliasId`). Which is why we'll need to run a migration script before deployment.

Reply:
> But both are breaking changes for Meteor pub/sub
Got it.

> But considering Holtzman is a cordova app, it doesn't matter. Old clients will automatically be upgraded to Apollos anyway and use all the new stuff.

Only if we push a new cordova build with Apollo in it, right? I guess it depends on what's local and what's loaded over the wire... are all cordova apps just wrappers pointing to a website, or are there local javascript/html on the device? I can't remember...it's been so long since I've done cordova stuff

Reply:
I think likes/following topics being moved into a collection in Heighliner is fine, we just need to support backwards compatibility with this `options` argument, which is just used to specify topics to follow, which doesn't depend on the db structure of likes/following, since they're just names...right?

Reply:
It's just a wrapper but it creates a bridge via javascript that opens access to native functions (like storage, sound, video, etc). I don't think we'll need to do anything special on our end but I can have a quick look.

Reply:
Likes in Holtzman go into an array stored in redux and are compared against whatever is rendered. The publisher searches for likes via `userId` which will change to `PrimaryAliasId` after we migrate. This means users logged in via the old authentication mechanism will have no way of knowing which likes belong to them.

Reply:
There are a couple of patches I could add though to make it know after the migration is done. Maybe also put the old `userId` in the likes collection

Reply:
I'm sure I can get backwards compatibility working but I'm not sure how long it will take

Reply:
@Lepozepo I'm only concerned about GraphQL schema backwards compatibility, TBH. If the old app is broken because it stores likes locally, w/e. As long as the schema guarantees backwards compatibility, I'm happy.

Reply:
Ok, I'll restore it as soon as saved payments is done. Unless it's more urgent?

Reply:
> urgent

Nope, not urgent. Sounds good!